src/server.c

```diff
@@ -1999,48 +1999,26 @@ static inline robj *createSharedString(const char *string) {
     return makeObjectShared(createStringObject(string, strlen(string)));
 }
 
-static inline robj *createSharedStringByPtr(void *ptr) {
-    return makeObjectShared(createObject(OBJ_STRING, ptr));
+static inline robj *createSharedStringByPtr(sds s) {
```

Comment:
ByPtr, rename to BySds or FromSds?

It takes ownership of the sds, so it can be seen as a function to convert from sds to robj. Therefore maybe FromSds is better?

Reply:
I suddenly realized that i used ptr because of this...
```
    for (j = 0; j < OBJ_SHARED_INTEGERS; j++) {
        shared.integers[j] = createSharedStringByPtr((void *)(long)j);
        initObjectLRUOrLFU(shared.integers[j]);
        shared.integers[j]->encoding = OBJ_ENCODING_INT;
    }
```

Reply:
yes, that is true, updated.